kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/lookup/EntryBoHelperServiceImpl.java

```diff
@@ -0,0 +1,323 @@
+
+            case KFSPropertyConstants.FINANCIAL_OBJECT_CODE:
+                retval.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, getFieldValue(row, KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR));
+                retval.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, getFieldValue(row, KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE));
+                retval.put(field, column.getPropertyValue());
+                return retval;
+
+            case KFSPropertyConstants.FINANCIAL_SUB_OBJECT_CODE:
+                if (StringUtils.equalsIgnoreCase(PdpPropertyConstants.CustomerProfile.CUSTOMER_DEFAULT_SUB_OBJECT_CODE, column.getPropertyValue())) {
+                    return retval;
+                }
+                retval.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, getFieldValue(row, KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR));
+                retval.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, getFieldValue(row, KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE));
+                retval.put(KFSPropertyConstants.ACCOUNT_NUMBER, getFieldValue(row, KFSPropertyConstants.ACCOUNT_NUMBER));
+                retval.put(KFSPropertyConstants.FINANCIAL_OBJECT_CODE, getFieldValue(row, KFSPropertyConstants.FINANCIAL_OBJECT_CODE));
+                retval.put(field, column.getPropertyValue());
+                return retval;
+
+            case KFSPropertyConstants.FINANCIAL_BALANCE_TYPE_CODE:
+            case KFSPropertyConstants.FINANCIAL_OBJECT_TYPE_CODE:
+            case KFSPropertyConstants.PROJECT_CODE:
+                if (StringUtils.equalsIgnoreCase("----------", column.getPropertyValue())) {
+                    return retval;
+                }
+                retval.put(KFSPropertyConstants.CODE, column.getPropertyValue());
+                return retval;
+
+            case KFSPropertyConstants.UNIVERSITY_FISCAL_PERIOD_CODE:
+                retval.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, getFieldValue(row, KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR));
+                retval.put(field, column.getPropertyValue());
+                return retval;
+
+            case KFSPropertyConstants.FINANCIAL_DOCUMENT_TYPE_CODE:
+            case KFSPropertyConstants.REFERENCE_FINANCIAL_DOCUMENT_TYPE_CODE:
+                DocumentTypeService documentTypeService = SpringContext.getBean(DocumentTypeService.class);
+                String docTypeId = documentTypeService.getIdByName(column.getPropertyValue());
+                retval.put(KFSPropertyConstants.DOCUMENT_TYPE_ID, docTypeId);
+                retval.put(KFSConstants.DOC_FORM_KEY, KFSConstants.DOC_FORM_KEY_88888888);
+                return retval;
+
+        }
+        LOG.debug("Error: Unknown field.");
+        return retval;
+    }
+
+    private static String getInquiryUrl() {
+        String inquiryUrl = KRADServiceLocator.getKualiConfigurationService().getPropertyValueAsString(KRADConstants.APPLICATION_URL_KEY);
+        String potentialUrlAddition = "kr/";

```

Comment:
"Potential" is a bit of a misnomer (and was what it was called in the class I stole this from). the inquiryURL received by the KualiConfigurationService has the potential to have a "/" at the end, and if it doesn't, it has to be added. I'm not sure what the "kr" stands for, but it's in the path for all of the Inquiry URLs that are generated.